### PR TITLE
Habitevent controller

### PR DESCRIPTION
**Pull request by j-vent**
Create a controller class for habit events which can talk with firebase. Has functions for creating and updating (and possibly reading) from habit_events collection.

**DON'T APPROVE: Unit Tests are not working. Functionality that works inside regular java activity classes is not authorized by JUnit (?) Might need to use firestore emulator**

## Description of Changes
Move create, update functions out of habit event activity
Create unit tests for each function

## Checklist

- [X] Automated tests
- [X] Screenshots included (if necessary)
- [X] Code is well commented

Closes #ISSUE_ID

### CODE REVIEW

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java [hunk 1]
```diff
@@ -9,25 +9,30 @@
...
     }
 
-    HabitEventModel(String location, Date date, String description) {
+    HabitEventModel(String location, Date date, String comment) {
         this.ID = "s1w9kwequ290"; // placeholder ID
         this.location = location;
-        this.description = description;
+        this.comment = comment;
         this.date = date;
         image = null;
```
Comment (cheetiantey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java:37]: Should be `this.image = null` :)

Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java:37]: added!

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java [hunk 2]
```diff
@@ -9,55 +9,181 @@
...
         this.location = location;
-        this.description = description;
+        this.comment = comment;
         this.date = date;
         this.image = image;
+        this.habitID = habitID;
     }
 
-    HabitEventModel(String location, Date date, String description) {
+    HabitEventModel(String location, Date date, String comment, String habitID) {
```
Comment (MacMalainey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java:56]: Add documentation for constructors please

**Review by MacMalainey — CHANGES_REQUESTED** [review]
LGTM other than documentation issues

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java (hunk 2 above)
Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java]: Added and removed unused contstructors.

**Review by MacMalainey — DISMISSED** [review]

**Review by loravocado — DISMISSED** [review]
Lgtm! :D

**Review by loravocado — APPROVED** [review]

**Review by MacMalainey — APPROVED** [review]

**Pull request merged**